devito/ops/node_factory.py

```diff
@@ -0,0 +1,214 @@
+
+        Returns
+        -------
+        :class:`Add`
+            Object of :class:`Add`.
+        """
+        return Add(lhs, rhs)
+
+    def new_mul_node(self, lhs, rhs):
+        """
+        Creates a new mathematical multiplication node operation.
+
+        Parameters
+        ----------
+        lhs
+            The left hand side of the multiplication.
+        rhs : [type]
+            The right hand side of the multiplication.
+
+        Returns
+        -------
+        :class:`Mul`
+            Object of :class:`Mul`.
+        """
+        return Mul(lhs, rhs)
+
+    def new_divide_node(self, num, den):
+        """
+        Creates a new division node operation.
+
+        Notes
+        -----
+        The division is represented by the multiplicating the numerator with 
+        denominator's inverse.
+
+        Parameters
+        ----------
+        num
+            The division numerator.
+        den : [type]
+            The division denominator.
+
+        Returns
+        -------
+        :class:`Mul`
+            Object of :class:`Mul` representing the division.
+        """
+        return Mul(num, Pow(den, Integer(-1)))
+
+    def new_grid(self, devito_indexed, dimensions):
+        """
+        Creates an :class:`Indexed` object using OPS representation.
+        If the pair grid name and time dimension was alredy created, it will return
+        the stored value associated with this pair.
+
+        Parameters
+        ----------
+        devito_indexed : :class:`Indexed`
+            Indexed object using devito representation.
+
+        dimensions : list
+            List of the dimensions alredy translated to OPS.
+
+        Returns
+        -------
+        :class:`Indexed`
+            Indexed object using OPS representation.
+        """
+
+        def getDimensionsDisplacement(dimensions):
+            disp = []
+
+            for dim in dimensions:
+                if dim.is_Add:
+                    lhs, = dim.as_two_terms()
+                    disp.append(str(lhs))
+                else:
+                    disp.append('0')
+
+            return disp
+
+        # Gets the dimension displacement.
+        disp = getDimensionsDisplacement(dimensions[1:])
+
+        # Builds the grid identifier.
+        grid_id = '%s%s' % (devito_indexed.name, dimensions[0])
+
+        if grid_id not in self.ops_grids:
+            # Creates the indexed object.
+            # Always will have one dimension that will be generated using the macro
+            # OPS_ACC but it will not represent the real dimension used, because we
+            # can have a 3 dimension problem and only one Dimension object.
+            grid = Array(name=grid_id,
+                         dimensions=[Dimension(name=namespace['ops_acc'])],
+                         dtype=devito_indexed.dtype)
+            
+            self.ops_grids[grid_id] = grid
+        else:
+            grid = self.ops_grids[grid_id]
```

Comment:
reminder: no need to call these "grids". These objects are Functions in core devito, right? even though here you cast them into array for code generation purposes

Reply:
I got your point, and you are right.
I'll use `ops_arg`, because it could the data associated with:  `ops_arg_dat`, `ops_arg_idx`, `ops_arg_gbl` ...

Reply:
maybe we should ask istvan's opinion on that?
i'd agree with dats although there seems to be a slight difference between `ops_dats` shat those args actually are

Reply:
i just saw your commit, `ops_arg` looks good indeed. 
yet wouldn't hurt to have instavan's feedback on that imo

Reply:
Actually i got this idea of name from Istvan code on `OPS`. I went to the OPS source code, and looked how they were calling this variables. Of course we can also confirm with him personally, just to be sure.